fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -425,6 +425,10 @@ public FHIRPersistenceJDBCImpl(Properties configProps, IConnectionProvider cp) t
                         return resultBuilder.success(false).build();
                     }
                 }
+                
+                if (searchContext.getPageSize() == 0) {
```

Comment:
Do you want to check page size or page number?
BTW, do we check the page size and number in the API layer to prevent them from being set to equal or less than 0?

Reply:
Yes, see validatePagingContext. We validate pageSize and pageNumber. The logic you have highlighted is to check pageSize after validation and if pageSize == 0, then set the summary parameter to COUNT (which is what the spec states is the correct behavior).

Reply:
I see, thanks for the clarification!